cms/admin/placeholderadmin.py

```diff
@@ -158,9 +158,51 @@ def edit_field(self, request, object_id, language):
 
 
 class PlaceholderAdminMixin(object):
+    class Meta:
+        warnings.warn("PlaceholderAdminMixin has been removed", DeprecationWarning)
 
-    def _get_attached_admin(self, placeholder):
-        return placeholder._get_attached_admin(admin_site=self.admin_site)
+
+class PlaceholderAdmin(admin.ModelAdmin):
+
+    def has_add_permission(self, request):
+        return False
+
+    def has_change_permission(self, request, obj=None):
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        return False
+
+    def has_module_permission(self, request):
+        return False
+
+    def log_addition(self, request, object, object_repr):
```

Comment:
Without the methods the admin creates log entries which causes the test_log_entries test case to fail because the placeholder operations are logged via signals and not the admin. They need to stay